k8s.gcr.io/Vanity-Domain-Flip.md

```diff
@@ -0,0 +1,191 @@
+
+## Prow Integration
+
+The act of invoking the promoter as a postsubmit against the k8s.io repo is done
+by [Prow][prow], as the `post-k8sio-cip` Prow job. There are other Prow jobs
+that integrate with the promoter, and the ones relevant to this doc are outlined
+in the table below:
+
+- [`post-k8sio-cip`](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml)
+  Postsubmit job against k8s.io repo holding promoter manifests. The promoter
+  manifests here are those that promote from the various staging subproject
+  repos to `gcr.io/k8s-artifacts-prod/<subproject>/<image>`. It uses the
+  `k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com` service
+  account to write to `{asia,eu,us}.gcr.io/k8s-artifacts-prod`. For all
+  intents and purposes, **this is the gatekeeper for new images going into
+  `k8s-artifacts-prod`**.
+- [`ci-k8sio-cip`](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml)
+  Like `post-k8sio-cip`, but runs periodically. This is to ensure
+  that even if images are accidentally deleted from
+  `gcr.io/k8s-artifacts-prod`, they are automatically copied back. It also
+  acts as a kind of sanity check, to ensure that the promoter can run at all.
+- [`ci-k8sio-backup`](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml)
+  Runs an hourly backup of all GCR images in
+  `{asia,eu,us}.gcr.io/k8s-artifacts-prod` to
+  `{asia,eu,us}.gcr.io/k8s-artifacts-prod-bak/YEAR/MONTH/DAY/HOUR/...`.
+- [`pull-k8sio-cip`](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/sig-release/cip/container-image-promoter.yaml)
+  Dry run version of `post-k8sio-cip`. It is run as a presubmit
+  check to any PR against [k8s.io Github repo][k8sio]. In particular, it
+  catches things like tag moves (which are disallowed). Unlike
+  `post-k8sio-cip`, it does not run in the trusted cluster, because it does
+  not need to use prod credentials (in fact, it doesn't use any creds).
+- [`pull-cip-e2e`](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/sig-release/cip/container-image-promoter.yaml)
+  Runs an E2E test for changes to the promoter source code. This
+  test checks that the promoter can promote images (its main purpose). It uses
+  the `k8s-infra-gcr-promoter@k8s-cip-test-prod.iam.gserviceaccount.com`
+  service account to use the `k8s-cip-test-prod` GCP project resources for its
+  test cases (creation/deletion of GCR images, etc.).
+- [`pull-cip-auditor-e2e`](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/sig-release/cip/container-image-promoter.yaml)
+  Like `pull-cip-e2e`, but runs E2E tests for the auditing
+  mechanism built into the promoter; this mechanism (known as "cip-auditor"),
+  runs as a Cloud Run service. It uses the
```

Comment:
runs in which project?

Reply:
where is code and config?

Reply:
Is this the e2e for auditor or the actual auditor?

Reply:
Ack